lib/datadog/appsec/metrics/telemetry.rb

```diff
@@ -17,6 +17,19 @@ def report_rasp(type, result)
           AppSec.telemetry.inc(namespace, 'rasp.rule.match', 1, tags: tags) if result.match?
           AppSec.telemetry.inc(namespace, 'rasp.timeout', 1, tags: tags) if result.timeout?
         end
+
+        def report_waf_requests(waf_metrics, request_milestones)
+          tags = {
+            waf_version: Datadog::AppSec::WAF::VERSION::BASE_STRING,
+            rule_triggered: (!waf_metrics.matches.zero?).to_s,
+            request_blocked: request_milestones.request_blocked.to_s,
+            rate_limited: request_milestones.request_rate_limited.to_s,
+            waf_error: (!waf_metrics.errors.zero?).to_s,
+            waf_timeout: (!waf_metrics.timeouts.zero?).to_s
+          }
+
+          AppSec.telemetry.inc(Ext::TELEMETRY_METRICS_NAMESPACE, 'waf.requests', 1, tags: tags)
+        end
```

Comment:
I think this is a wrong move. An existing telemetry is sent on every call, where this method is more exporter of metrics via telemetry that is suppose to be called once. And IMHO it should be reflected via module name and the method name.

I don't think that this placement is correct. Technically we have here RaspTelemetry which is about rasp. In existing Context you can see that I have `record_rasp` and then `export_rasp_metrics`. Same could be done here - recording and exporting. And yes, exporting is done via Telemetry, not span Metrics functionality